packages/components/src/components/SquiggleChart.tsx

```diff
@@ -266,29 +22,30 @@ export interface SquiggleChartProps {
   environment?: environment;
   /** If the result is a function, where the function starts, ends and the amount of stops */
   chartSettings?: FunctionChartSettings;
-  /** When the environment changes */
-  onChange?(expr: squiggleExpression): void;
+  /** When the squiggle code gets reevaluated */
+  onChange?(expr: squiggleExpression | undefined): void;
```

Comment:
What does the undefined case here mean?

Reply:
From the top post:

> onChange gets called with an undefined value on errors, it wasn't called on errors before (which I think is good, errors should trickle down instead of being hidden)

Previously, if you changed the code to something which doesn't compile then observable would keep using an old cached version in other cells dependent on it. Now it'll give you undefined which would trickle down and cause errors in dependent cells.

I think this is better because it's more consistent with how observable behaves in case of other kinds of errors.